### ISSUE TITLE
Address online only prebuilts

### ISSUE THREAD

**Issue by MichaelSimons** [dotnet/source-build#3557]
The following prebuilts are being reported in the online builds.  These do not appear in offline builds.

[Online Build Leg](https://dev.azure.com/dnceng/internal/_build/results?buildId=2224020&view=logs&j=941f2e24-48a0-594a-6b1a-0e16aaeb8606&t=cffff4a5-f52c-50c6-45c9-52a0bd619b26) (Microsoft internal link)

```
/vmr/repo-projects/Directory.Build.targets(688,5): warning : 3 new packages used not in baseline! See report at /vmr/artifacts/prebuilt-report/baseline-comparison.xml for more information. Package IDs are: [/vmr/repo-projects/dotnet.proj]
/vmr/repo-projects/Directory.Build.targets(688,5): warning : Microsoft.CodeAnalysis.Analyzers.3.3.4 [/vmr/repo-projects/dotnet.proj]
/vmr/repo-projects/Directory.Build.targets(688,5): warning : Microsoft.CodeAnalysis.Common.4.7.0-3.23314.3 [/vmr/repo-projects/dotnet.proj]
/vmr/repo-projects/Directory.Build.targets(688,5): warning : Microsoft.CodeAnalysis.CSharp.4.7.0-3.23314.3 [/vmr/repo-projects/dotnet.proj]
```

**Comment by mthalman** [comment]
This regression was caused by a CodeAnalysis dependency change in aspnetcore (https://github.com/dotnet/aspnetcore/pull/49245, https://github.com/dotnet/aspnetcore/pull/48817).

The reason it wasn't caught by prebuilt detection at the repo level is because the version of the loaded CodeAnalysis packages (4.7.0) are coming from roslyn's source build intermediates. Since they come from intermediates, they're excluded from the list of prebuilts. But in the full source-build, the version of CodeAnalysis built by roslyn is 4.8.0. This version gets flowed to aspnetcore via PVP but aspnetcore is [pinned to 4.7.0](https://github.com/dotnet/aspnetcore/pull/48817/files#diff-1ea18ff65faa2ae6fed570b83747086d0317f5e4bc325064f6c14319a9c4ff67R248). So aspnetcore loads 4.7.0, causing the prebuilt.

I still need to investigate why no failure occurred in full source-build's offline build or stage 2 build.

**Comment by mthalman** [comment]
> I still need to investigate why no failure occurred in full source-build's offline build or stage 2 build.

Both the offline build and stage 2 build have the behavior of NuGet automatically resolving the package version to a higher version. In both cases, `4.7.0-3.23314.3` is defined as the minimum in the version range (i.e. `[4.7.0-3.23314.3, )`). For the offline build, NuGet resolves this to `4.7.0-3.23328.4` which comes from previously source-built artifacts. For the bootstrapping scenario, stage 1 produces version `4.8.0-1.23370.8` and, in stage 2, NuGet resolves `4.7.0-3.23314.3` to that 4.8.0 version.
